compiler/src/dotty/tools/MainGenericRunner.scala

```diff
@@ -166,9 +166,10 @@ object MainGenericRunner {
           .withTargetScript(arg)
           .withScriptArgs(tail*)
       else
-        val newSettings = if arg.startsWith("-") then settings else settings.withPossibleEntryPaths(arg).withModeShouldBePossibleRun
-        process(tail, newSettings.withResidualArgs(arg))
-
+        val modalSettings = if arg.startsWith("-") then settings else settings.withPossibleEntryPaths(arg).withModeShouldBePossibleRun
+        val newSettings = modalSettings.withResidualArgs(arg)
+        process(tail, newSettings)
```

Comment:
This change does not seem to be needed to make `tests/run-with-compiler/i14541.scala` pass. Why was this added? How could I test this?

Reply:
Just refactors the expression so I could println settings. I can revert.

Reply:
We need a test tool that proves a local refactor cannot change behavior.